generic/proof-script.el

```diff
@@ -2025,7 +2025,16 @@ This function expects the buffer to be activated for advancing."
 	(lastpos   (nth 2 (car semis)))
 	(vanillas  (proof-semis-to-vanillas semis displayflags)))
     (proof-script-delete-secondary-spans startpos lastpos)
-    (proof-extend-queue lastpos vanillas)))
+    (proof-extend-queue lastpos vanillas))
+
+  ;; arm the timeout timer
+  ;; cancelled 1. in `proof-shell-exec-loop' unless `proof-shell-busy' or
+  ;; 2. in the case of error, in `proof-shell-error-or-interrupt-action'
+  (if proof-shell-timeout-warn
+      (setq proof-shell-timer
+          (run-with-timer proof-shell-timeout-warn nil
+                          'message "This command is taking a while. \
+Is it malformed? Do C-c C-c to interrupt prover or C-c C-x to terminate it."))))
```

Comment:
For the message, please use something like
```
(substitute-command-keys "bla bla \\[proof-interrupt-process] bla bla")
```
This way the actual shortcut will be displayed, even if it has changed from the default one.